posts/apache-spark-3.0.md

```diff
@@ -0,0 +1,54 @@
+
+PySpark errors are not always user-friendly to Python users because Spark was originally designed to run on a JVM. This release will simplify PySpark exceptions by making them more Pythonic and will also avoid showing unnecessary JVM stack trace.
+
+# Spark SQL engine
+
+Most Spark applications are using the Spark SQL engine (based on Databricks stats) and one of the goal of this release was to improve performances and ANSI compatibility. Spark 3.0 will perform around 2x faster than a Spark 2.4 environment in the total runtime. In order to improve performances and query tuning a new framework was introduced: [Adaptive Query Execution (AQE)](https://databricks.com/blog/2020/05/29/adaptive-query-execution-speeding-up-spark-sql-at-runtime.html).
+
+AQE will generate a better execution plan at runtime which is crucial because of the separation between storage and compute in Spark (which make data unpredictable). In other words, AQE will create a better plan even if your data are absent or inaccurate (the initial Spark plan would have been inadequate).
+
+The ANSI compatibility aims to make workload migration from other SQL engines easier by improving the SQL compliance. Spark will forbid you the use of SQL reserved keywords (LIKE, IN, SELECT, etc) to avoid having issues later on. Also, when inserting data in a predefined schema, Spark will validate the data types to improve data quality. The property `spark.sql.ansi.enabled` must be set to `true` in order to enable this feature.
+
+# Other updates
+
+Data streaming is a big part in every big data use cases for its near-real-time processing. Structured streaming was initially introduced in Spark 2.0 and now a new [Web UI](https://spark.apache.org/docs/3.0.0/web-ui.html#structured-streaming-tab) is available to inspect your streaming jobs with two set of statistics (aggregate information of jobs completed and detailed information about queries).
+
+This major release is the result of contributions from over 440 contributors which demonstrate the power of a strong open-source community. It is also good to note that the contributors are part of some very big organization like Google, Microsoft, Facebook, Netflix or Databricks.
+
+More details about this release can be found on [the release notes here](https://spark.apache.org/releases/spark-release-3-0-0.html). You can also consult the [Jira project](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12315420&version=12339177) for a list of all the tickets included in the release.
+
+# How to use this version
+
+In order to test out this new release, you will have to use Databricks since this version is currently not available on the main cloud providers (like AWS EMR). You can sign up for a [14 days trial account here](https://databricks.com/try-databricks). You can also install Spark locally (with [Homebrew](https://brew.sh/) and `brew install apache-spark` on a Mac) and play with the Spark Shell.
```

Comment:
```suggestion
In order to test out this new release, you will have to use Databricks since this version is currently not available on the main cloud providers (like AWS EMR). You can sign up for a [14 day trial account here](https://databricks.com/try-databricks). You can also install Spark locally (with [Homebrew](https://brew.sh/) and `brew install apache-spark` on a Mac) and play with the Spark Shell.
```